composable_kernel/include/tensor_operation/gridwise_2d_reduction_blockwise.hpp

```diff
@@ -0,0 +1,863 @@
+                               p_dst_global,
+                               ws_indices_global,
+                               indices_global);
+    }
+    else
+    {
+        GridwiseReduction::RunSecondCallWithIndices(in2dDesc,
+                                                    out1dDesc,
+                                                    in_elementwise_op,
+                                                    acc_elementwise_op,
+                                                    alpha,
+                                                    p_src_global,
+                                                    beta,
+                                                    p_dst_global,
+                                                    ws_indices_global,
+                                                    indices_global);
+    };
+};
+
+template <typename InDataType,
+          typename OutDataType,
+          typename AccDataType,
+          typename In2dDescType,
+          typename Out1dDescType,
+          typename ReduceOperation,
+          typename InElementwiseOperation,
+          typename OutElementwiseOperation,
+          bool PropagateNan,
+          bool BetaIsZero,
+          index_t BlockSize,
+          index_t MThreadClusterSize,
+          index_t KThreadClusterSize,
+          index_t MThreadSliceSize,
+          index_t KThreadSliceSize,
+          index_t InSrcVectorDim,
+          index_t InSrcVectorSize,
+          index_t OutDstVectorSize>
+struct GridwiseReduction_xy_to_x_blockwise
+{
+    static constexpr bool reorder_thread_cluster = (InSrcVectorDim == 0);
+
+    static constexpr auto buffer1dDesc =
+        make_naive_tensor_descriptor_packed(make_tuple(Number<BlockSize>{}));
+
+    using blockwise_reduce = PartitionedBlockwiseReduction_1d_block_buffer<decltype(buffer1dDesc),
+                                                                           BlockSize,
+                                                                           MThreadClusterSize,
+                                                                           KThreadClusterSize,
+                                                                           reorder_thread_cluster,
+                                                                           ReduceOperation,
+                                                                           PropagateNan>;
+
+    template <typename T>
+    using PassThroughOp = tensor_operation::element_wise::UnaryIdentic<T, T>;
+
+    static constexpr auto I0 = Number<0>{};
+
+    static constexpr index_t M_BlockTileSize = MThreadClusterSize * MThreadSliceSize;
+    static constexpr index_t K_BlockTileSize = KThreadClusterSize * KThreadSliceSize;
+
+    using Accumulation =
+        detail::accumulate_with_nan_check<PropagateNan, ReduceOperation, AccDataType>;
+
+    __device__ static void Run(const In2dDescType& in2dDesc,
+                               const Out1dDescType& out1dDesc,
```

Comment:
```
const InGridDesc_M_K& in_grid_desc_m_k,
const OutGridDesc_M& out_grid_desc_m,
```

Please also apply name changes in other places too
https://github.com/ROCmSoftwarePlatform/composable_kernel/wiki/Coding-Style#tensor-descriptor

Reply:
> ```
> const InGridDesc_M_K& in_grid_desc_m_k,
> const OutGridDesc_M& out_grid_desc_m,
> ```
> 
> Please also apply name changes in other places too
> 
> https://github.com/ROCmSoftwarePlatform/composable_kernel/wiki/Coding-Style#tensor-descriptor

Good !